libz-rs-sys/src/gz.rs

```diff
@@ -535,6 +535,67 @@ pub unsafe extern "C-unwind" fn gzerror(file: gzFile, errnum: *mut c_int) -> *co
     }
 }
 
+/// Clear the error and end-of-file state for `file`.
+///
+/// # Arguments
+///
+/// * `file` - A gzip file handle, or null
+///
+/// # Safety
+///
+/// `file`, if non-null, must be an open file handle obtained from [`gzopen`] or [`gzdopen`].
+#[cfg_attr(feature = "export-symbols", export_name = crate::prefix!(gzclearerr))]
+pub unsafe extern "C-unwind" fn gzclearerr(file: gzFile) {
+    // Get internal structure and check integrity
+    let Some(state) = (unsafe { file.cast::<GzState>().as_mut() }) else {
+        return;
+    };
+    if state.mode != GzMode::GZ_READ && state.mode != GzMode::GZ_WRITE {
+        return;
+    }
+
+    // Clear error and EOF
+    if state.mode == GzMode::GZ_READ {
+        state.eof = false;
+        state.past = false;
+    }
+
+    // Safety: we've checked state, and gz_error supports a null message argument.
+    unsafe { gz_error(state, Z_OK, ptr::null()) };
```

Comment:
not for this PR perhaps, but could gz_error use something like this for the message?

```rust
enum Msg {
    None,
    Errcode(c_int),
    Static(&'static str),
}
```

To just reduce pointers/c strings.

Reply:
Definitely! This will cover all the ways zlib-ng uses gz_error:
```
enum Msg {
    None,
    Errcode((c_int, &str))
}
```
or equivalently:
```
Option<(c_int, &str)>
```

I'm working on an update to this PR to add more test cases, and I'll include that change.